onnx/external_data_helper.py

```diff
@@ -134,17 +141,25 @@ def convert_model_to_external_data(
                     "location must be a relative path that is relative to the model path."
                 )
             file_name = location
+        current_offset = 0
         for tensor in tensors:
+            tensor_size = len(tensor.raw_data)
             if (
                 tensor.HasField("raw_data")
-                and sys.getsizeof(tensor.raw_data) >= size_threshold
+                and tensor_size >= size_threshold
             ):
-                set_external_data(tensor, file_name)
+                if align_offset and tensor_size > align_threshold:
+                    # Align to the larger of the page size or the allocation granularity
+                    alignment_factor = max(4096, allocation_granularity)  # Assuming 4k page size
+                    current_offset = (current_offset + alignment_factor - 1) // alignment_factor * alignment_factor # Align to the next page or alloc granularity boundary
+
+                set_external_data(tensor, file_name, offset=current_offset)
+                current_offset += tensor_size
     else:
         for tensor in tensors:
             if (
                 tensor.HasField("raw_data")
-                and sys.getsizeof(tensor.raw_data) >= size_threshold
+                and len(tensor.raw_data) >= size_threshold
```

Comment:
This looks reasonable ... but I wonder why they used `sys.getsizeof` originally? Are the two guaranteed to return the same value in all cases?

Reply:
sys.getsizeof()  in Python apparently returns the size of the object, not the actual length of the tensor raw data. I kept hitting a bug with the previous code where it would write out too big of the data + padding vs what the size of the tensor was originally. This is because of this incorrect length calculation. I think nobody noticed the bug before because it was "close enough" to use as a size_threshold

Reply:
for (a) I would expect this would work with multiple files but I admit I did not test that. Code review should tell us the answer